relay-server/src/utils/managed_envelope.rs

```diff
@@ -43,23 +43,34 @@ impl Handling {
     }
 }
 
+/// Represents the decision on whether or not to keep an envelope item.
+pub enum RetainItem {
+    /// Keep the item.
+    Keep,
+    /// Drop the item and log an outcome for it.
+    Drop(Outcome, DataCategory, usize), // TODO: derive DataCategory and usize from Item
+    /// Drop the item without logging an outcome.
+    DropSilently,
```

Comment:
As a follow-up, would it make sense to document under what circumstances we should use each of these?